platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/TransactionPoolNexus.java

```diff
@@ -148,19 +152,35 @@ public synchronized boolean submitApplicationTransaction(@NonNull final SwirldTr
             return false;
         }
 
-        if (transaction.getSize() > maximumTransactionSize) {
+        int transactionSize = transactionConversion(transaction).getSize();
+        if (transactionSize > maximumTransactionSize) {
             // FUTURE WORK: This really should throw, but to avoid changing existing API this will be changed later.
             illegalTransactionLogger.error(
                     EXCEPTION.getMarker(),
                     "transaction has {} bytes, maximum permissible transaction size is {}",
-                    transaction.getSize(),
+                    transactionSize,
                     maximumTransactionSize);
             return false;
         }
 
         return submitTransaction(transaction, false);
     }
 
+    /**
+     * Convert a transaction to a {@link ConsensusTransactionImpl}. This is a hack because {@code OneOf<>} interface
+     * can not return a size of the transaction.
+     *
+     * @param transaction the transaction to convert
+     * @return the converted transaction
+     */
+    private ConsensusTransactionImpl transactionConversion(final OneOf<PayloadOneOfType> transaction) {
```

Comment:
Question: If I understand it correctly, the OneOf Contains an ENUM type to define the transaction type as key and the raw bytes of the transaction as value. If that is correct: why do you wrap the bytes in a new object and return it to get it's lenght later? Wouldn't it be better to create getSize(OneOf...) method that extracts the value of the OneOf and returns the size?

Reply:
... and now I see that you use it at other places, to to get the serialized length. Based on that it makes sense ;)

Reply:
But: If we need to transform the OneOf anyway at this position to a concrete transaction class (and as it looks this method might be called more than 1 time per transaction) doesn't it make sense to do that directly at the start and work with the concrete transaction instead of OneOf? The benefit of OneOf is less  memory + object creation at runtime. Since we now create a new object we miss all benefits. Or do I miss something?

Reply:
The grater picture is that all self serializable objects will go away and the "raw' OneOfs will be used.